src/server/dragonfly_test.cc

```diff
@@ -758,21 +758,72 @@ TEST_F(DflyEngineTest, Bug496) {
 TEST_F(DefragDflyEngineTest, TestDefragOption) {
   // Fill data into dragonfly and then check if we have
   // any location in memory to defrag. See issue #448 for details about this.
-  EXPECT_GE(max_memory_limit, 100'000);
-  const int NUMBER_OF_KEYS = 184000;
+  max_memory_limit = 300'000;             // control memory size so no need for too many keys
+  constexpr int kNumberOfKeys = 100'000;  // this fill the memory
+  constexpr int kKeySize = 137;
+  const uint64_t kMinExpectedDefragCount = 80;
+
+  std::vector<std::string> keys2delete;
+  keys2delete.push_back("del");
+
+  // Generate a list of keys that would be deleted
+  // The keys that we will delete are all in the form of "key-name:1<other digits>"
+  // This is because we are populating keys that has this format, but we don't want
+  // to delete all keys, only some random keys so we deleting those that start with 1
+  constexpr int kFactor = 10;
+  int kMaxNumKeysToDelete = 10'000;
+  int current_step = kFactor;
+  for (int i = 1; i < kMaxNumKeysToDelete; current_step *= kFactor) {
+    for (int j = i - 1 + current_step; i < current_step; i++, j++) {
```

Comment:
what does this loop do? should it be `i < current_step` or `j < current_step` ?

Reply:
j is just to set the string value, it is not controlling the loop number of iterations - you can write this without the j at all it would just looking at the line that set the name of the key as something like  i + current_step - 1. So j is just to make it less verbose there. But you are using i for both loops.

Reply:
ok, I understand now but it does break my pattern recognition. 

Do you mind writing the equivalent code with inner loop like:
``for (; i < current_step; ++i) {
  int j = i - 1 + current_step;
  ....
}
```

Reply:
NP